https://issues.redhat.com/browse/ACM-26213 note about fail-over (#8417)
open-cluster-management/rhacm-docs: governance/gatekeeper_operator/gk_operator_overview.adoc

- Evaluate Kubernetes resource compliance requests for the Kubernetes API by using the Gatekeeper constraints.
- Use OPA as the policy engine and use Rego as the policy language.

> *Prerequisite:* You need a {acm} or {ocp} Plus subscription to install Gatekeeper and apply Gatekeeper policies to your cluster.

Reviewer: Why is the subscription requirement being removed? It seems to me it's still relevant?

Author: It is removed because this is not a task file. I will add it to the file where it applies. making a few more commits

Reviewer: Sure, I get that it's not a task file, but it does seem more relevant to me in the overview since users would acquire the subscription before looking at how to install or configure it, right? So maybe this content can just move under the new "Important" heading?

Author: I moved it to the General Support topic. We can do a live review so that I can explain my changes.

Gatekeeper policy support:: Red Hat supports only the Gatekeeper Operator and does not support individual Gatekeeper policies. Support ensures that the operator is functioning.

> Default fail-open behavior:: By default, the `failurePolicy` parameter for Gatekeeper is set to `Ignore`. If the Gatekeeper webhook becomes unavailable, constraints are not enforced. To learn about changing the `failurePolicy` when you configure the {gate}, see _Configuring the {gate}_ in the list of resources. For more information about configuring fail-open and fail-closed behavior, see link:https://open-policy-agent.github.io/gatekeeper/website/docs/howto/failopen/[Gatekeeper documentation].

Author: ```suggestion
Default fail-open behavior:: By default, the `failurePolicy` parameter for Gatekeeper is set to `Ignore`. If the Gatekeeper webhook becomes unavailable, constraints are not enforced. To learn about changing the `failurePolicy` when you configure the {gate}, see _Configuring the {gate}_ in the list of resources. For more information about configuring fail-open and fail-closed behavior, see link:https://open-policy-agent.github.io/gatekeeper/website/docs/failing-closed[Gatekeeper documentation].
```

- Evaluate Kubernetes resource compliance requests for the Kubernetes API by using the Gatekeeper constraints.
- Use OPA as the policy engine and use Rego as the policy language.

*Important:*

> Gatekeeper policy support:: Red Hat supports only the Gatekeeper Operator. Individual Gatekeeper policies are not supported. Support ensures that the operator is functioning.

Reviewer 2: Do we need the double colon? If so, avoid my suggestion
```suggestion
Gatekeeper policy support: Red Hat supports only the Gatekeeper Operator. Individual Gatekeeper policies are not supported. Support ensures that the operator is functioning.
```

Author: We need to double colon for the descriptions.

> Gatekeeper policy support:: Red Hat supports only the Gatekeeper Operator. Individual Gatekeeper policies are not supported. Support ensures that the operator is functioning.

Reviewer 2: Are the individual policies not supported by Red Hat? If so, let's remove passive voice, like IBM SG suggests when we can: 

```suggestion
Gatekeeper policy support:: Red Hat supports only the Gatekeeper Operator and does not support individual Gatekeeper policies. Support ensures that the operator is functioning.
```

> Default fail-open behavior:: By default, Gatekeeper is configured to fail open. If the Gatekeeper webhook becomes unavailable, requests to the Kubernetes API are not blocked. For more information about configuring fail-open and fail-closed behavior, see link:https://open-policy-agent.github.io/gatekeeper/website/docs/howto/failopen/[Gatekeeper documentation].

Reviewer 3: Could you also add that users can configure spec.webhook.failurePolicy: Fail in the Gatekeeper CR?

Author: ```suggestion
Default fail-open behavior:: By default, Gatekeeper is configured to fail open. If the Gatekeeper webhook becomes unavailable, requests to the Kubernetes API are not blocked. To learn about configuring the {gate}, see _Configuring the {gate}]. For more information about configuring fail-open and fail-closed behavior, see link:https://open-policy-agent.github.io/gatekeeper/website/docs/howto/failopen/[Gatekeeper documentation].
```

> Default fail-open behavior:: By default, Gatekeeper is configured to fail open. If the Gatekeeper webhook becomes unavailable, requests to the Kubernetes API are not blocked. To learn about configuring the {gate}, see _Configuring the {gate}]. For more information about configuring fail-open and fail-closed behavior, see link:https://open-policy-agent.github.io/gatekeeper/website/docs/howto/failopen/[Gatekeeper documentation].

Author: ```suggestion
Default fail-open behavior:: By default, Gatekeeper is configured to fail open. If the Gatekeeper webhook becomes unavailable, requests to the Kubernetes API are not blocked. To learn about configuring the {gate}, see _Configuring the {gate}_. For more information about configuring fail-open and fail-closed behavior, see link:https://open-policy-agent.github.io/gatekeeper/website/docs/howto/failopen/[Gatekeeper documentation].
```

> Default fail-open behavior:: By default, Gatekeeper is configured to fail open. If the Gatekeeper webhook becomes unavailable, requests to the Kubernetes API are not blocked. To learn about configuring the {gate}, see _Configuring the {gate}_. For more information about configuring fail-open and fail-closed behavior, see link:https://open-policy-agent.github.io/gatekeeper/website/docs/howto/failopen/[Gatekeeper documentation].

Reviewer 2: "fail open" is not a phrase I'm usd to. it also seems a little colloquial. Do you mean any of these meanings? 

Gatekeeper is configured to fail. 
Gatekeeper is configured to instantly fail. 
Gatekeeper is configured to fail and let you know when it does. 

These are ideas. It seems to me that "open" conveys that it's obvious to the user when Gatekeeper fails, and there's nothing hidden or unexpected about its failure.

Reviewer 2: Later, I see you have "fail open" behavior. So if that's a term that Gatekeeper users would be familiar with, keep it :)

Author: Based on the Gatekeeper docs, it seems like it will be familiar to the user. I agree that it is a bit confusing. In the Gatekeeper docs, there is a specific parameter named `failurePolicy`. When `failurePolicy` is set to `Ignore`, constraints are not enforced when the webhook is unreachable.

Let me add that to help

> Default fail-open behavior:: By default, Gatekeeper is configured to fail open. If the Gatekeeper webhook becomes unavailable, requests to the Kubernetes API are not blocked. To learn about configuring the {gate}, see _Configuring the {gate}_. For more information about configuring fail-open and fail-closed behavior, see link:https://open-policy-agent.github.io/gatekeeper/website/docs/howto/failopen/[Gatekeeper documentation].

Reviewer 2: Is the italicized Doc part of the link you provide? If so, that could be easier for users to find if we re-write it to something like the following: 

```suggestion
Default fail-open behavior:: By default, Gatekeeper is configured to fail open. If the Gatekeeper webhook becomes unavailable, requests to the Kubernetes API are not blocked. To learn about configuring the {gate} and the fail-open and fail-closed behavior, see _Configuring the {gate}_ in the link:https://open-policy-agent.github.io/gatekeeper/website/docs/howto/failopen/[Gatekeeper documentation].
```

Author: It is italicized because the doc is already linked within this file. Christine wanted me to let the user know that they can change the `failurePolicy` parameter. From my perspective, we shouldn't be adding a task to the file bc it is more of an overview. I thought it be best to just link to the doc where the task is listed. Again, let me make that more clear in the doc
